releases/cosmwasm/2025-01-Stellar-GMP-v1.0.0.md

```diff
@@ -0,0 +1,274 @@
+```
+
+3. Instantiate `MultisigProver`
+
+```bash
+node ./cosmwasm/deploy-contract.js instantiate -c MultisigProver --fetchCodeId --instantiate2
+
+```
+
+4. Set environment variables
+
+- Network-specific environment variables: These variables need to be updated by the network.
+
+```bash
+VOTING_VERIFIER=$(cat ./axelar-chains-config/info/$DEVNET.json | jq ".axelar.contracts.VotingVerifier[\"$CHAIN\"].address" | tr -d '"')
+GATEWAY=$(cat ./axelar-chains-config/info/$DEVNET.json | jq ".axelar.contracts.Gateway[\"$CHAIN\"].address" | tr -d '"')
+MULTISIG_PROVER=$(cat ./axelar-chains-config/info/$DEVNET.json | jq ".axelar.contracts.MultisigProver[\"$CHAIN\"].address" | tr -d '"')
+MULTISIG=$(cat ./axelar-chains-config/info/$DEVNET.json | jq .axelar.contracts.Multisig.address | tr -d '"')
+REWARDS=$(cat ./axelar-chains-config/info/$DEVNET.json | jq .axelar.contracts.Rewards.address | tr -d '"')
+```
+
+- General environment variables
+
+```bash
+RUN_AS_ACCOUNT=[wasm deployer key address]
+DEPOSIT_VALUE=100000000
+REWARD_AMOUNT=1000000uamplifier
+```
+
+5. Register stellar gateway at the Router
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Router \
+  -t "Register Gateway for stellar" \
+  -d "Register Gateway address for stellar at Router contract" \
+  --deposit $DEPOSIT_VALUE \
+  --runAs $RUN_AS_ACCOUNT \
+  --msg "{
+    \"register_chain\": {
+      \"chain\": \"$CHAIN\",
+      \"gateway_address\": \"$GATEWAY\",
+      \"msg_id_format\": \"hex_tx_hash_and_event_index\"
+      }
+    }"
+```
+
+6. Register chain on ampd
+
+```bash
+for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-chain-support validators $CHAIN ; done
+
+for i in $(seq 0 4); do kubectl exec -it ampd-set-2-axelar-amplifier-worker-"$i" -n $ENV -c ampd -- ampd register-public-key ed25519 ; done
+```
+
+7. Register prover contract on coordinator
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Coordinator \
+  -t "Register Multisig Prover for stellar" \
+  -d "Register Multisig Prover address for stellar at Coordinator contract" \
+  --deposit $DEPOSIT_VALUE \
+  --runAs $RUN_AS_ACCOUNT \
+  --msg "{
+    \"register_prover_contract\": {
+      \"chain_name\": \"$CHAIN\",
+      \"new_prover_addr\": \"$MULTISIG_PROVER\"
+    }
+  }"
+```
+
+8. Authorize Stellar Multisig prover on Multisig
+
+```bash
+node cosmwasm/submit-proposal.js execute \
+  -c Multisig \
+  -t "Authorize Multisig Prover for stellar" \
+  -d "Authorize Multisig Prover address for stellar at Multisig contract" \
+  --runAs $RUN_AS_ACCOUNT \
+  --deposit $DEPOSIT_VALUE \
+  --msg "{
+    \"authorize_callers\": {
+      \"contracts\": {
+        \"$MULTISIG_PROVER\": \"$CHAIN\"
+      }
+    }
+  }"
+```
+
+9. Update verifier set
+
+```bash
+axelard tx wasm execute $MULTISIG_PROVER '"update_verifier_set"' --from amplifier --gas auto --gas-adjustment 1.2
+```
+
+### Setup reward pools
```

Comment:
Can reward pools be setup before the update verifier set step (can ask Edwin)? Reorder if so to better pipeline it

Reply:
seems that there is no requirement for the verifier set to be updated before the rewards pool is created. I don't think the order matters as long as both are executed, otherwise gmp will fail. @milapsheth you think is better to create the pool first? probably makes sense in mainnet so that the pools can be created in parallel before the chain is operation

Reply:
Thanks, yep, should reduce latency wherever possible